### ISSUE TITLE
Example of Super Resolution

### ISSUE THREAD

**Issue by hetong007** [apache/incubator-mxnet#777]
Besides the [Neural Art](https://github.com/dmlc/mxnet/tree/master/example/neural-style) example, I propose to make another interesting demo to do Super Resolution. The purpose of this demo is to enlarge a 2D anime plot with noise reduction.

One available online demo is http://waifu2x.udp.jp/
Its source: https://github.com/nagadomi/waifu2x

**Comment by winstywang** [comment]
It is still an active research area. I saw several papers on arxiv last week. I think we should investigate which paper to implement first.

**Comment by piiswrong** [comment]
This is cool. Is anyone familiar with lua?

**Comment by piiswrong** [comment]
BTW we should try to get some pr with our neural style example

**Comment by antinucleon** [comment]
The reason why we don't PR neural art is: 1. CuDNN bug. 2. I think if we add bind on multi-device, it will be more powerful as we can generate higher resolution images, which all other alternatives can't.

**Comment by brightown** [comment]
I recommand ：
http://arxiv.org/abs/1511.04587 
http://arxiv.org/abs/1504.06993

**Comment by huangzehao** [comment]
Hi! I am trying to do this work. 

I have implemented a simple super-resolution work in MXNet. Though the result is better than simple interpolation method (bicubic), but it's not a competitive result compare to state-of-the-art result.

The work I am trying to implement is http://arxiv.org/abs/1511.04587, a 20 layers convolution neural work. I have implemented it in caffe [caffe-vdsr](https://github.com/huangzehao/caffe-vdsr) with learning rate 0.1 and clip_gradient 0.1 and get a good result.  There is also a torch implementation with learning rate 0.1 and clip_gradient 0.025

But in MXNet, it seems difficult to get a convergent result with the same parameters in caffe-vdsr and torch implementation.  I change clip_gradient to 0.025 because of the different implementation of clip_gradient in caffe and MXNet.

I am newbie in MXNet and I have two questions about the setting of clip_gradient:
(1) Is the implementation of clip_gradient in MXNet the same as in torch?
(2) How to check the usage of clip_gradient? I set monitor to show gradients, but they are not in range [-clip_gradient, clip_gradient]

Thank you!
